### DISCUSSION TITLE
mysql 8.4 is not support in laragon?

### DISCUSSION THREAD

**Discussion by alwiismail66** [leokhoa/laragon#774] [Q&A]
i try download mysql 8.4 and then try change the version mysql in laragon but i got error then when i try back to version 8.0 i got error too how to resolve it ?  is log from mysqld.log

2024-05-08T12:26:47.237100Z 0 [Warning] [MY-010097] [Server] Insecure configuration for --secure-file-priv: Current value does not restrict location of generated files. Consider setting it to a valid, non-empty path.
2024-05-08T12:26:47.238637Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
2024-05-08T12:26:47.238692Z 0 [System] [MY-010116] [Server] C:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld (mysqld 8.0.30) starting as process 14036
2024-05-08T12:26:47.323572Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-05-08T12:26:47.791724Z 1 [ERROR] [MY-013171] [InnoDB] Cannot boot server version 80030 on data directory built by version 80400. Downgrade is not supported
mysqld: Can't open file: 'mysql.ibd' (errno: 0 - )
2024-05-08T12:26:47.847260Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine
2024-05-08T12:26:47.848922Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
2024-05-08T12:26:47.849590Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-05-08T12:26:47.856149Z 0 [System] [MY-010910] [Server] C:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld: Shutdown complete (mysqld 8.0.30)  MySQL Community Server - GPL.
2024-05-08T12:26:57.137330Z 0 [Warning] [MY-010097] [Server] Insecure configuration for --secure-file-priv: Current value does not restrict location of generated files. Consider setting it to a valid, non-empty path.
2024-05-08T12:26:57.137382Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
2024-05-08T12:26:57.137413Z 0 [System] [MY-010116] [Server] C:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld (mysqld 8.0.30) starting as process 13032
2024-05-08T12:26:57.157622Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-05-08T12:26:57.581546Z 1 [ERROR] [MY-013171] [InnoDB] Cannot boot server version 80030 on data directory built by version 80400. Downgrade is not supported
mysqld: Can't open file: 'mysql.ibd' (errno: 0 - )
2024-05-08T12:26:57.613727Z 1 [ERROR] [MY-010334] [Server] Failed to initialize DD Storage Engine
2024-05-08T12:26:57.614921Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
2024-05-08T12:26:57.615258Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-05-08T12:26:57.617814Z 0 [System] [MY-010910] [Server] C:\laragon\bin\mysql\mysql-8.0.30-winx64\bin\mysqld: Shutdown complete (mysqld 8.0.30)  MySQL Community Server - GPL.

**Comment by meorajrul** [comment]
see my guide here https://gist.github.com/meorajrul/b57803bf1b4ddfd2f93e6ad37c3ac5f2

**Reply by AWqxKAWERbXo** [reply]
@meorajrul would it be ok to move your guide to the https://github.com/leokhoa/laragon/wiki section?

**Reply by Pen-y-Fan** [reply]
I've updated the wiki with a link to @meorajrul 's guide 👍

**Comment by luisAntonioLAGS** [comment]
I was checking it, and apparently laragon does not support version 8.4 of MySQL. I think we will have to wait for an update from Laragon.

**Comment by fpierrat** [comment]
I updated to 8.4 and got it working thanks to @meorajrul and a few other sources. Still I have a problem.
For tests reasons, I need to use sometimes 8.0 and sometimes 8.4. 

Since I launched mysqld 8.4, I can't have laragon start 8.0 any more.
The problem is (or at least seems to be), Laragon keeps creating a data\mysql-8 folder and trying to work from there (both for mysqld 8.0 and 8.4), totally ignoring the separate data\mysql-8.0 and data\mysql-8.4 data folders I configured in bin\mysql-8.0.18-winx64\my.ini and bin\mysql-8.4.0-winx64\my.ini

I'm not the only one having this issue, I also found it described here: https://forums.mysql.com/read.php?10,710182,710182#msg-710182

Note: I can start mysqld manually from the command line, but then I'm losing all start-and-stop-and-change-version-and-start-again in Laragon's UI.

Note 2: making my.ini readonly doesn't change anything.

Can someone tell me where this data\mysql-8 folder is configured and how to disable it so Laragon just lets mysqld use the folder specified in the my.ini file?

**Reply by ppfoong** [reply]
It is in Laragon's Preference, data directory.

"mysql-8" will be appended to the data directory.

Current Laragon has issue running 8.4 due to it always add a line "default_authentication_plugin=mysql_native_password" in my.ini but that config is not accepted by MySQL 8.4. It is still working fine with MySQL 8.0.37.

**Reply by fpierrat** [reply]
So this is what I feared, "mysql-8" is hard-coded in laragon, no conf nowhere.
"default_authentication_plugin=mysql_native_password" is not a problem, you just have to set the my.ini settings as you want them to be, then set the file as readonly (file properties) => laragon can't change the file, so mysqld starts without the "default_authentication_plugin" error, everything works... (see meorajrul's link above)
BUT IT WORKS IN MYSQL-8 FOLDER, so all my databases get upgraded by 8.4 and then I can't run 8.0 anymore because mysqld refuses to start with DB files that have been upgraded to 8.4... :'-(
Short version: I really have no problems running 8.4, my problem is switching from and back to 8.0 / 8.4 alternatively depending on my testing needs.

**Reply by AWqxKAWERbXo** [reply]
Fun Fact: MAMP is still on MySQL 5.x

**Reply by AWqxKAWERbXo** [reply]
@fpierrat did you try to run laragon and laragon portable in parallel? So you could constantly use both with different settings / mysql versions?

If you'd use different ports, they could coexist and run at the same time.

**Reply by fpierrat** [reply]
I finally re-initialized everything with 8.0 and will forget 8.4 for now.
8.0 is acutally the one used in production, I'll stick to that for my tests for now...

**Reply by FlavioSuarez** [reply]
Hi,

Maybe you could have 2 folders into the data folder: mysql-8.0 and the default mysql-8 for the latest version...
Copy the mysql-8 folder to mysql-8.0
Then edit the my.ini on the MySQL 8.0 folter, set datadir= "C:/laragon6/data/mysql-8.0", save and turn it read-only

Hope this helps...
